internal/pkg/deploy/pipeline.go

```diff
@@ -94,25 +104,77 @@ type BitbucketSource struct {
 	RepositoryURL string
 }
 
+// PipelineSourceFromManifest processes manifest info about the source based on provider type.
+// The return boolean is true for CodeStar Connections sources that require a polling prompt.
+func PipelineSourceFromManifest(source *manifest.Source) (interface{}, bool, error) {
+	switch source.ProviderName {
+	case ghV1ProviderName:
+		return &GitHubV1Source{
+			ProviderName:                ghV1ProviderName,
+			Branch:                      (source.Properties["branch"]).(string),
+			RepositoryURL:               GitHubURL((source.Properties["repository"]).(string)),
+			PersonalAccessTokenSecretID: (source.Properties["access_token_secret"]).(string),
+		}, false, nil
+	case ghProviderName:
+		// If the creation of the user's pipeline manifest predates Copilot's conversion to GHv2/CSC, the provider
+		// listed in the manifest will be "GitHub," not "GitHubV1." To differentiate it from the new default
+		// "GitHub," which refers to v2, we check for the presence of a secret, indicating a v1 GitHub connection.
+		if source.Properties["access_token_secret"] != nil {
+			return &GitHubV1Source{
+				ProviderName:                ghV1ProviderName,
+				Branch:                      (source.Properties["branch"]).(string),
+				RepositoryURL:               GitHubURL((source.Properties["repository"]).(string)),
+				PersonalAccessTokenSecretID: (source.Properties["access_token_secret"]).(string),
+			}, false, nil
+		} else {
+			return &GitHubSource{
+				ProviderName:  ghProviderName,
+				Branch:        (source.Properties["branch"]).(string),
+				RepositoryURL: GitHubURL((source.Properties["repository"]).(string)),
+			}, true, nil
+		}
+	case ccProviderName:
+		return &CodeCommitSource{
+			ProviderName:  ccProviderName,
+			Branch:        (source.Properties["branch"]).(string),
+			RepositoryURL: (source.Properties["repository"]).(string),
+		}, false, nil
+	case bbProviderName:
+		return &BitbucketSource{
+			ProviderName:  bbProviderName,
+			Branch:        (source.Properties["branch"]).(string),
+			RepositoryURL: (source.Properties["repository"]).(string),
+		}, true, nil
+	default:
+		return nil, false, fmt.Errorf("invalid repo source provider: %s", source.ProviderName)
+	}
+}
+
+type GitHubSource struct {
```

Comment:
doc comments plz 🙏 we can explain how this one represents a codestar connection whereas v1 is the weird one with access tokens

Reply:
Yes! Sorry!

(Poor v1. How quickly it fell from being the only one, the normal one, the right one, the star...to being "the weird one"!)

Reply:
who you calling the weird one